app/assets/javascripts/components/modal.js.jsx.erb

```diff
@@ -1,6 +1,7 @@
 var Modal = React.createClass({
   getInitialState: function() {
-    return { show: false };
+    return { show: false,
+            showCloseButton : this.props.showCloseButton === undefined ? true : this.props.showCloseButton};
```

Comment:
We may want to loosely check against `null` to accept both `null` and `undefined`:

```js
showCloseButton : this.props.showCloseButton == null ? true : this.props.showCloseButton};
```

Or just reverse the logic:

```js
showCloseButton: this.props.showCloseButton !== false
```